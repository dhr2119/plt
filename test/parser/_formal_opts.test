INT MAIN LPAREN RPAREN LBRACE RETURN SEMI RBRACE FLOAT ID LPAREN INT ID RPAREN LBRACE RBRACE
