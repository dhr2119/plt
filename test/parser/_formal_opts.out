ACCEPT
[program:
  [mdecl:
    INT
    MAIN
    LPAREN
    RPAREN
    LBRACE
    [vdecl_list:]
    [stmt_list: [stmt_list:] [stmt: RETURN SEMI]]
    RBRACE
  ]
  [fdecls:
    [fdecls:]
    [fdecl:
      [primitives: FLOAT]
      ID
      LPAREN
      [formals_opt: [formal_list: [primitives: INT] ID]]
      RPAREN
      LBRACE
      [vdecl_list:]
      [stmt_list:]
      RBRACE
    ]
  ]
  EOF
]
